compiler/src/dotty/tools/repl/Rendering.scala

```diff
@@ -122,7 +116,26 @@ private[repl] class Rendering(parentClassLoader: Option[ClassLoader] = None) {
         else
           s
       }
-  }
+
+  /** Rewrap value classes to their's Wrappers and evaluate their `toString` value.
+   *
+   * @param sym Value Class symbol
+   * @param value underlying value
+   */
+  def rewrapValueClass(sym: Symbol, value: Object)(using Context): Option[Object] =
+    if (sym.isValueClass && !sym.isPrimitiveValueClass) then
+      val valueClassName = sym.flatName.encode.toString
+      val valueClass = Class.forName(valueClassName, true, classLoader())
+
+      for {
+        constructor    <- valueClass.getConstructors.headOption
+        toStringMethod <- valueClass.getMethods.find(_.getName == nme.toString_.toString)
+      } yield {
+        val instance = constructor.newInstance(value)
+        toStringMethod.invoke(instance)
```

Comment:
I think you can skip calling toString and just return the instance since replStringOf will call toString itself.